src/snowflake/cli/api/entities/common.py

```diff
@@ -39,3 +40,21 @@ def perform(self, action: EntityActions, action_ctx: ActionContext):
         Performs the requested action.
         """
         return getattr(self, action)(action_ctx)
+
+
+class SqlExecutor(SqlExecutionMixin):
```

Comment:
the mixin has state, no? I'm very reluctant to introduce yet another stateful global var. We could either 1) replace the mixin everywhere, or 2) make the base executor stateless, and make the mixin reuse the base executor.

Reply:
Maybe it would be simpler to override and blank the stateful mixin behavior here? (it's basically a snowpark session)

Reply:
Wouldn't flipping the inheritance hierarchy basically achieve the same thing? The advantage then is that we can phase out the mixin and delete it over time, no need to change the executor after that.

Reply:
Yeah, you're right. I'll do it this way.